v4/newrelic/segments.go

```diff
@@ -214,18 +216,58 @@ func (s *DatastoreSegment) End() {
 	if s.StartTime.isEnded() {
 		return
 	}
+
+	s.addAttributes(s.StartTime.Span.SetAttribute)
 	s.StartTime.Span.SetName(s.name())
 	s.StartTime.end()
 }
 
+func (s *DatastoreSegment) addAttributes(set func(string, interface{})) {
+	set("db.system", valOrUnknown(string(s.Product)))
+	set("db.statement", valOrUnknown(s.statement()))
+	set("db.operation", valOrUnknown(s.Operation))
+
+	if net.ParseIP(s.Host) != nil {
+		set("net.peer.ip", s.Host)
+	} else {
+		set("net.peer.name", valOrUnknown(s.Host))
+	}
+	if s.PortPathOrID != "" {
+		if port, err := strconv.Atoi(s.PortPathOrID); err == nil {
+			set("net.peer.port", port)
+		}
+	}
+
+	switch s.Product {
+	case DatastoreCassandra:
+		set("db.cassandra.keyspace", valOrUnknown(s.DatabaseName))
+	case DatastoreRedis:
+		set("db.redis.database_index", valOrUnknown(s.DatabaseName))
+	case DatastoreMongoDB:
+		set("db.mongodb.collection", valOrUnknown(s.DatabaseName))
+	default:
+		set("db.name", valOrUnknown(s.DatabaseName))
+	}
+}
```

Comment:
Please check the otel spec to make sure I got all my logic correct. https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md